pkg/elasticsearch/indices.go

```diff
@@ -8,8 +8,36 @@ import (
 	"github.com/openshift/elasticsearch-operator/pkg/logger"
 	estypes "github.com/openshift/elasticsearch-operator/pkg/types/elasticsearch"
 	"github.com/openshift/elasticsearch-operator/pkg/utils"
+
+	"github.com/mitchellh/mapstructure"
 )
 
+func (ec *esClient) GetIndex(name string) (*estypes.Index, error) {
+	payload := &EsRequest{
+		Method: http.MethodGet,
+		URI:    name,
+	}
+	logger.DebugObject("GetIndex for %q", name)
+	ec.fnSendEsRequest(ec.cluster, ec.namespace, payload, ec.k8sClient)
+	if payload.Error != nil {
+		return nil, payload.Error
+	}
+	if payload.StatusCode == 404 {
+		return nil, nil
+	}
+	if payload.StatusCode != 200 && payload.StatusCode != 201 {
+		return nil, fmt.Errorf("failed getting index %s. Error code: %v, %v", name, payload.StatusCode != 200, payload.ResponseBody)
+	}
+
+	index := &estypes.Index{}
+	err := mapstructure.Decode(payload.ResponseBody, index)
```

Comment:
is this essentially unmarshalling the data into the `estypes.Index` type?

Reply:
No this decodes the unmarshalled `map[string]interface{}` to an actual type `estypes.Index` here.